demo_umami_content/src/Plugin/Block/UmamiDisclaimer.php

```diff
@@ -18,14 +19,68 @@ class UmamiDisclaimer extends BlockBase {
    * {@inheritdoc}
    */
   public function defaultConfiguration() {
-    return ['label_display' => FALSE];
+    return [
+      'label_display' => FALSE,
+      'umami_disclaimer' => [
+        'value' => '',
+        'format' => '',
+      ],
+      'umami_copyright' => [
+        'value' => '',
+        'format' => '',
+      ],
+    ] + parent::defaultConfiguration();
   }
 
   /**
    * {@inheritdoc}
    */
   public function build() {
-    return ['#markup' => '<span class="umami-disclaimer">' . $this->t('<strong>Umami Magazine & Umami Publications</strong> is a fictional magazine and publisher for illustrative purposes only.') . '</span><span class="umami-copyright">© 2017 ' . $this->t('Terms & Conditions') . '</span>'];
+    $disclaimer_html = check_markup($this->configuration['umami_disclaimer']['value'], $this->configuration['umami_disclaimer']['format']);
+    $copyright_html = check_markup($this->configuration['umami_copyright']['value'], $this->configuration['umami_copyright']['format']);
+
+    return [
+      '#type' => 'inline_template',
+      '#template' => '<span class="umami-disclaimer">{{ disclaimer }}</span><span class="umami-copyright">{{ copyright }}</span>',
+      '#context' => [
+        'disclaimer' => $disclaimer_html,
+        'copyright' => $copyright_html,
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockForm($form, FormStateInterface $form_state) {
+    $form = parent::blockForm($form, $form_state);
+
+    $form['umami_disclaimer'] = array(
+      '#type' => 'text_format',
+      '#title' => $this->t('Umami Disclaimer'),
+      '#default_value' => $this->configuration['umami_disclaimer']['value'],
+      '#format' => $this->configuration['umami_disclaimer']['format'],
+    );
+
+    $form['umami_copyright'] = array(
+      '#type' => 'text_format',
+      '#title' => $this->t('Umami Copyright Text'),
+      '#default_value' => $this->configuration['umami_copyright']['value'],
+      '#format' => $this->configuration['umami_copyright']['format'],
+    );
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockSubmit($form, FormStateInterface $form_state) {
+    parent::blockSubmit($form, $form_state);
+    $values = $form_state->getValues();
+
+    $this->configuration['umami_disclaimer'] = $values['umami_disclaimer'];
+    $this->configuration['umami_copyright'] = $values['umami_copyright'];
```

Comment:
Instead of assigning $values & using that, can we use _$form_state->getValue()_, i.e.:

`
parent::blockSubmit($form, $form_state);
$this->configuration['umami_disclaimer'] = $form_state->getValue('umami_disclaimer');
$this->configuration['umami_copyright'] = $form_state->getValue('umami_copyright');
`